test/luatest_helpers/server.lua

```diff
@@ -190,4 +190,28 @@ function Server:start(opts)
     getmetatable(getmetatable(self)).start(self, opts)
 end
 
+-- Eval a string similar to `Server:eval()` and reconnect to the server
+-- in case the connection has been dropped.
+function Server:eval_reconnect(code, ...)
+    if self.net_box == nil then
+        error('net_box is not connected', 2)
+    end
+    local ok, err = pcall(self.net_box.eval, self.net_box, code, ...)
+    if not ok and tostring(err):find('Peer closed') then
+        self:wait_until_ready()
+    elseif not ok then
+        error(err)
+    end
+    return err
```

Comment:
It looks a bit strange: the error is either raised or returned depending of whether the reconnect was occured.

Reply:
Right, renamed `err` to `res`.